File: GT668/Samples/Python/DataStorage.py
from GT668Driver import *


gt = GT668Driver()
gt.initialize(0)

# size of array per channel
size = 10

# read timeout
timeout = 10.0

# configure inputs
gt.set_input_prescale(GT668.Signal.GT_SIG_A, GT668.Prescale.GT_DIV_1)
gt.set_input_threshold(GT668.Signal.GT_SIG_A, GT668.ThresholdMode.GT_THR_VOLTS, 0.0)

# configure inputs
gt.set_input_prescale(GT668.Signal.GT_SIG_B, GT668.Prescale.GT_DIV_1)
gt.set_input_threshold(GT668.Signal.GT_SIG_B, GT668.ThresholdMode.GT_THR_VOLTS, 0.0)

#read tags up to 100 tags on either channel
gt.start_measurements()

# initialize timetags set object
timetagsSet = TimetagsSet(channel0Size=size, channel1Size=size)

# read time
start_time = gt.read_sys_time()

# the measurement will last untill the timeout
while gt.read_sys_time() - start_time < timeout:
    gt.read_timetags(timetagsSet)

    # when both channels have the specified tags count, terminates
    if timetagsSet.channel0Count == timetagsSet.channel0Size and timetagsSet.channel1Count == timetagsSet.channel1Size:
        break

#free card
gt.close()

#store as time aligned two row csv
GT668DataUtils.save_tags_to_time_aligned_two_collumns_CSV(timetagsSet, ',', "ch0", "ch1", "/path/to/simple_two_column.csv")

#store as simple csv
GT668DataUtils.save_tags_as_simple_csv(timetagsSet, True, "/path/to/simple.csv", ",")

#store as simple txt
GT668DataUtils.save_tags_as_simple_text(timetagsSet, True, "/path/to/simple.txt")

#store as formatted file
dff = Data_Format_Factory()

dff.file_type = File_Type.csv               #optional: default CSV
dff.tags_per_file = 50                      #optional: default -1 (save all tags in single file)
dff.delimiter = ';'                         #optional: default ','
dff.empty_tag_representation = "n/a"        #optional: default "---"
dff.file_name = "Some_file_name"            #optional: default GT668Tags
dff.restart_numbering_in_new_file = True    #optional: default false
dff.include_row_numbering = True            #optional: default true
dff.header = "HEADER"                       #optional: default None (no header)
dff.store_header_in_each_file = True        #optional: default false

#cell initialization
c1 = Cell()
c2 = Cell()
c3 = Cell()
c4 = Cell()

#single cell consists of prefix Var [either tags from channel 0 or 1] and suffix
#each cell can have all or just one of above fields
#cells are separated by Data_Format_Factory.delimiter field value

c1.prefix = "prefix"

c2.prefix = "pre_before_tag_value: "
c2.var = Var.ch_0_tags

c3.var = Var.ch_1_tags

c4.suffix = "suffix"

#row definition
dff.row = [c1, c2, c3, c4]                  #required: sets row definition, see documentation for details

GT668DataUtils.save_tags_with_formatting(timetagsSet, dff, "/path/to/")

File: GT668/Samples/Java/src/com/guidetech/gt668/samples/dataStorage/DataStorage.java
package com.guidetech.gt668.samples.dataStorage;

import java.io.IOException;

import com.guidetech.driver.gt668.api.IGT668Slr;
import com.guidetech.driver.gt668.api.exceptions.NoCellsDefinedException;
import com.guidetech.driver.gt668.api.exceptions.RowNotDefinedException;
import com.guidetech.driver.gt668.commons.GT668DataUtils;
import com.guidetech.driver.gt668.data.TimetagsSet;
import com.guidetech.driver.gt668.data.utils.Cell;
import com.guidetech.driver.gt668.data.utils.DataFormatFactory;
import com.guidetech.driver.gt668.data.utils.FileType;
import com.guidetech.driver.gt668.data.utils.Row;
import com.guidetech.driver.gt668.data.utils.Var;

public class DataStorage {

	static IGT668Slr gt;
	
	static int deviceNumber = 0;
	
	static double timeout = 2.0;
	
	public static void main(String[] args) throws IOException, ClassNotFoundException, RowNotDefinedException, NoCellsDefinedException {
		//initialize the board
	    gt.initialize(deviceNumber);


	    /**
	     * Configure card here
	     * Calibrate
	     */
	  
	    gt.startMeasurements();                    

	    //read tags
		double start_time = gt.readSysTime();
		TimetagsSet tags = new TimetagsSet();
		while(gt.readSysTime() - start_time < timeout) {

			tags.addTags(
					gt.readTimetags(15, 15)
					);
			//read up to 100 tags on either channel
			if(tags.getChannel0Count() >= 100 || tags.getChannel1Count() >= 100)
				break;
		}
		
		//free card
		gt.close();
		
		//include header?
		boolean includeHeader = true;
		
		//store as simple csv
		String simpleCSVPath = "/path/to/simple.csv";
		String delimiter = ",";
		GT668DataUtils.saveTagsAsSimpleCSV(tags, includeHeader, simpleCSVPath, delimiter);
		
		//store as simple txt
		String simpleTXTPath = "/path/to/simple.txt";
		GT668DataUtils.saveTagsAsSimpleText(tags, includeHeader, simpleTXTPath);
		
		//store as time aligned two row csv
		String ch0representation = "ch0";
		String ch1representation = "ch1";
		String twoRowCSVPath = "/path/to/two_row.csv";
		char delimiterCSV = ',';
		GT668DataUtils.saveTagsToTimeAlignedTwoColumnsCSV(tags, delimiterCSV, ch0representation, ch1representation, twoRowCSVPath);
		
		//store as bytes
		String bytesPath = "/path/to/bytes.ser";
		GT668DataUtils.saveTagsAsBytes(tags, bytesPath);
		
		//read bytes
		TimetagsSet retrievedTags = GT668DataUtils.readTagsFromBytes(bytesPath);
		
		//store tags with complex formatting
		//Data format object defines file structure
		DataFormatFactory dfe = new DataFormatFactory();
		dfe
			.setOutputFileType(FileType.CSV)					//optional: default CSV
			.setDelimiter(';') 									//optional: default ','
			.setFileName("FormattedFile")						//optional: default GT668Tags
			.setEmptyTagRepresentation("n/a")					//optional: default "---"
			.setIncludeRowNumbering(false)						//optional: default true
			.setMaxTagsPerFile(5)								//optional: default -1 (save all tags in single file)
			.setHeader("channel_0_tag;channel_1_tag")			//optional: default null (no header)
			.setStoreHeaderInEachFile(true)						//optional: default false
			.setRestartNumberingInNewFile(true)					//optional: default false
			.setRowDefinition((new Row())						//required: sets row definition, see documentation for details
										.addCell(new Cell("", "", Var.CH_0_TAG)) // This row definition has 4 cells, each cell has following structure:
										.addCell(new Cell("", "", Var.CH_1_TAG)) // String prefix + Var + String suffix
										.addCell(new Cell("someText"))			 // Single string constructor should be used for string only cells
										.addCell(new Cell("otherText")));		 // String Var consturctor should be used for String prefix + Var var cell
																				 // Var String constructor should be used for Var var + String suffix cell
																				 // String String Var constructor should be used for String prefix + Var var + String suffix cell
																			     // this row definition will result with following line:
																				 // Channel_0_tag Channel_1_tag someText otherText e.g. : 0.12345 0.67899 someText otherText
		String pathToFormattedFile = "/path/to/";
		GT668DataUtils.saveTagsWithFormatting(dfe, tags, pathToFormattedFile);
		

	}

}
